readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -68,86 +69,112 @@ function attach_elastic_search_query(data) {
+                                        }
                                     }
                                 }
 
-                                section_subtitle.html(section_subtitle_link);
-                                contents.append(section_subtitle);
-                                contents.append(section_content);
-                                contents.find('em').addClass('highlighted');
+                                contents.append(
+                                    $u.template(
+                                        section_template,
+                                        {
+                                            section_subtitle_link: section_subtitle_link,
+                                            section_subtitle: section_subtitle,
+                                            section_content: section_content
+                                        }
+                                    )
+                                );
                             }
 
                             // if the result is a sphinx domain object
                             if (inner_hits[j].type === "domains") {
 
                                 var domain = inner_hits[j];
-                                var domain_subtitle = $('<div class="rtd_search_subtitle">');
-                                var domain_subtitle_link = $('<a href="' + link + "#" + domain._source.anchor + '">');
-                                var domain_content = $('<span>');
+                                var domain_subtitle = domain._source.role_name;
+                                var domain_subtitle_link = link + "#" + domain._source.anchor;
+                                var domain_content = "";
 
                                 if (
                                     typeof domain._source.display_name === "string" &&
-                                    domain._source.display_name.length >= 2
+                                    domain._source.display_name.length >= 1
                                 ) {
-                                    domain_subtitle_link.html(
-                                        "(" +
-                                        domain._source.role_name +
-                                        ")" +
-                                        domain._source.display_name
-                                    );
-                                } else {
-                                    domain_subtitle_link.html(domain._source.role_name);
+                                    domain_subtitle = "(" + domain._source.role_name + ") " + domain._source.display_name;
                                 }
 
                                 // preparing domain_content
-                                domain_content.append(domain._source.type_display + " -- ");
+                                // domain_content = type_display --
+                                domain_content = domain._source.type_display + " -- ";
```

Comment:
This feels weird. Should this be a template also?

Reply:
I have reduced the `ifs` here.
It is less complicated and more readable now.